File: Editorial/2211_Count_Collisions_on_a_Road/JavaScript/2_Counting.js
var countCollisions = function (directions) {
    const n = directions.length;
    let l = 0,
        r = n - 1;

    while (l < n && directions[l] === "L") {
        l++;
    }

    while (r >= l && directions[r] === "R") {
        r--;
    }

    let res = 0;
    for (let i = l; i <= r; i++) {
        if (directions[i] !== "S") {
            res++;
        }
    }
    return res;
};

File: Editorial/2201-2300/2211_Count_Collisions_on_a_Road/Go/2_Counting.go
func countCollisions(directions string) int {
	n := len(directions)
	l, r := 0, n-1

	for l < n && directions[l] == 'L' {
		l++
	}

	for r >= l && directions[r] == 'R' {
		r--
	}

	res := 0
	for i := l; i <= r; i++ {
		if directions[i] != 'S' {
			res++
		}
	}
	return res
}
